mobile-browser-based-version/src/Task Definition/lus_covid.js

```diff
@@ -0,0 +1,250 @@
+        
+        return preprocessed_data
+    }
+
+
+    async loadLocalImage(filename) {
+        return new Promise((res, rej) => {
+            var img = new Image();
+            img.src = filename
+            img.width = this.training_information.IMAGE_W
+            img.height = this.training_information.IMAGE_H
+            img.onload = () =>{
+                var output = tf.browser.fromPixels(img)
+                res(output)
+            }
+        });
+    }
+
+
+    async image_preprocessing(src){
+        const tensor = await this.loadLocalImage(src)
+        
+        const batched = tensor.reshape([this.training_information.IMAGE_H, this.training_information.IMAGE_W, 3])
+
+        const processedImg = batched.toFloat().div(127).sub(1).expandDims(0);
+
+        let representation = net.predict(processedImg)
+
+        return representation
+    }
+
+    labels_preprocessing(labels){
+        const nb_labels = labels.length
+        const labels_one_hot_encoded = []
+        for(let i = 0; i< labels.length; ++i){
+            labels_one_hot_encoded.push(this.one_hot_encode(labels[i]))
+        }
+        
+        console.log(labels_one_hot_encoded)
+        return tf.tensor2d(labels_one_hot_encoded, [nb_labels, this.training_information.NUM_CLASSES])
+    }
+
+    mean_tensor(tensors){
+        let result = tensors[0]
+
+        for(let i = 1; i< tensors.length; ++i){
+            const tensor = tensors[i]
+            result = result.add(tensor)
+        }
+        result = result.div(tf.scalar(tensors.length))
+
+        return result
+    }
+
+    one_hot_encode(label){
+        const result = []
+        for (let i = 0; i < this.training_information.NUM_CLASSES; i++){
+            if(this.training_information.LABEL_LIST[i]==label){
+                result.push(1)
+            }else{
+                result.push(0)
+            }
+        }
+        return result
+    }
+
+    /**
+     * Get all training data as a data tensor and a labels tensor.
+     *
+     * @returns
+     *   xs: The data tensor, of shape `[numTrainExamples, 28, 28, 1]`.
+     *   labels: The one-hot encoded labels tensor, of shape
+     *     `[numTrainExamples, 2]`.
+     */
+    async getTrainData(image_uri, labels_preprocessed, image_names) {
+        const dict_images = {}
+        const dict_labels = {}
+        let patients = new Set()
+
+        for(let i = 0; i<image_names.length; ++i){
+            const id = parseInt(image_names[i].split("_")[0])
+            patients.add(id)
+
+            let res = []
+
+            if(id in dict_images){ 
+                res = dict_images[id]
+            }else{
+                dict_labels[id] = labels_preprocessed[i]
+            }
+
+            res.push(await this.image_preprocessing(image_uri[i]))
+
+            dict_images[id] = res
+        }
+
+
+        console.log("Number of patients found was of "+Object.keys(dict_images).length)
+
+        let image_tensors1 = {}
```

Comment:
do we need the **1** at the end of the name? Try to avoid non-descriptive variable names